File: src/main/java/com/itheima/springboot_01_ssm_startquick/entity/Order.java
package com.itheima.springboot_01_ssm_startquick.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

/**
 * (Order)实体类
 *
 * @author woshihaoren
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Order implements Serializable {
    /**
     * 订单id
     */
    private Long id;
    /**
     * 关联user表的openId
     */
    private String openId;
    /**
     * 项目名称
     */
    private String proName;
    /**
     * 预约时间
     */
    private String makeDate;
    /**
     * 预约客户姓名
     */
    private String userName;
    /**
     * 预约状态
     */
    private String orderState;
    /**
     * 下单时间
     */
    private String placeDate;
    /**
     * 手机号
     */
    private String userTell;
    /**
     * 留言
     */
    private String information;
    /**
     * 关联商家id
     */
    private Long busId;
    /**
     * 关联产品id
     */
    private Long proId;
    
    private Long tecId;

    private Project project;

    private Business business;


}

File: Test1/miniprogram/pages/order/order.ts
// pages/order/order.ts
Page({

  /**
   * 页面的初始数据
   */
  data: {
      httpurl:'http://localhost:82/',
      imageurl:'http://localhost:82/static/',
      project:{},
      date:'2020-09-15',
      time:'09:00'
  },


  /**
   * 
   * 改变日期的函数
   */
  bindDateChange:function(e){
    console.log("日期改变函数：",e)
    this.setData({
      date:e.detail.value
    })
  },

  /**
   * 
   * 改变时间的函数
   */
  bindTimeChange:function(e){
    this.setData({
      time:e.detail.value
    })
  },


/**
 * 确认预约按钮触发事件
 */

formSubmit:function(e){

     var that = this;
     wx.getStorage({
       key:'user',
       success:(res)=>{
         console.log(res)
         //获取成功证明有token,是登录状态,可以预约
         wx.request({
           url:that.data.httpurl + 'order/createOrder',
           data:{
            token: res.data.token,
            proName: that.data.project.proName,
            makeDate: that.data.date+ " "+ that.data.time,
            userName: e.detail.value.username,
            userTell: e.detail.value.usertell,
            information: e.detail.value.information,
            busId: that.data.project.busId,
            proId: that.data.project.id
           },
           success:(result)=> {
             console.log(result)
             if(result.data.code == 3) {
              wx.showToast({
                title: result.data.message,
                icon: 'success',
                duration: 2000
              })
              // 跳转
              setTimeout(() => {
                wx.switchTab({
                  url: '/pages/index/index'
                })
              }, 2000);
             }
           }
         })
       }
     })
},

  /**
   * 生命周期函数--监听页面加载
   */
  onLoad(option) {
    
    var that = this;
    //请求项目的数据进行页面填充
     wx.request({
       url:that.data.httpurl + 'project/byId',
       data:{
        id:option.proid
       },
       //成功就返回数据
       success:(res)=>{
         that.setData({
           project: res.data.data
         })
       }
     }) 
  },

  /**
   * 生命周期函数--监听页面初次渲染完成
   */
  onReady() {

  },

  /**
   * 生命周期函数--监听页面显示
   */
  onShow() {

  },

  /**
   * 生命周期函数--监听页面隐藏
   */
  onHide() {

  },

  /**
   * 生命周期函数--监听页面卸载
   */
  onUnload() {

  },

  /**
   * 页面相关事件处理函数--监听用户下拉动作
   */
  onPullDownRefresh() {

  },

  /**
   * 页面上拉触底事件的处理函数
   */
  onReachBottom() {

  },

  /**
   * 用户点击右上角分享
   */
  onShareAppMessage() {

  }
})